pkg/tfbridge/config_encoding.go

```diff
@@ -84,50 +69,35 @@ func (*configEncoding) convertStringToPropertyValue(s string, typ shim.ValueType
 	return resource.NewPropertyValue(jsonValue), nil
 }
 
-// Compute an approximation of which properties should be secret. A property should be secret if it is statically marked
-// as secret in the schema, or has nested sub-properties marked in this way. A property should also be secret if the
-// dynamic value sent from the program is marked as secret.
-func (enc *configEncoding) ComputeSecrets(props *structpb.Struct,
-	opts plugin.MarshalOptions) map[resource.PropertyKey]struct{} {
-	opts.KeepSecrets = true
-
-	secrets := make(map[resource.PropertyKey]struct{})
-
-	for k := range enc.sensitiveFields {
-		secrets[k] = struct{}{}
-	}
-
-	pm, err := enc.UnmarshalProperties(props, opts)
-	if err == nil {
-		for k, v := range pm {
-			if v.ContainsSecrets() {
-				secrets[k] = struct{}{}
-			}
-		}
+func (*ConfigEncoding) zeroValue(typ shim.ValueType) resource.PropertyValue {
+	switch typ {
+	case shim.TypeBool:
+		return resource.NewPropertyValue(false)
+	case shim.TypeInt, shim.TypeFloat:
+		return resource.NewPropertyValue(0)
+	case shim.TypeList, shim.TypeSet:
+		return resource.NewPropertyValue([]interface{}{})
+	default:
+		return resource.NewPropertyValue(map[string]interface{}{})
 	}
-	return secrets
 }
 
-// Ensure that the sensitive top-level properties are marked as secret.
-func (enc *configEncoding) MarkSecrets(secrets map[resource.PropertyKey]struct{},
-	pm resource.PropertyMap) resource.PropertyMap {
-	copy := make(resource.PropertyMap)
-	for k, v := range pm {
-		_, secret := secrets[k]
-		if secret {
-			copy[k] = resource.MakeSecret(v)
-		} else {
-			copy[k] = v
-		}
+func (enc *ConfigEncoding) unmarshalOpts() plugin.MarshalOptions {
+	return plugin.MarshalOptions{
```

Comment:
Inline MarshalOptions into ConfigEncoding as it's not really configurable but does a specific thing.